src/librados/IoCtxImpl.cc

```diff
@@ -148,13 +148,11 @@ struct C_aio_notify_Complete : public C_aio_linger_Complete {
     if (ret_val == 0 && r < 0) {
       ret_val = r;
     }
+    lock.Unlock();
 
     if (acked && finished) {
-      lock.Unlock();
```

Comment:
`acked` and `finished` are updated with `lock`, so probably we should check them with the same lock?

Reply:
Indeed -- you're correct. Thx.